VMware.CloudFoundation.CertificateManagement.psm1

```diff
@@ -1319,6 +1319,494 @@ Function Install-EsxiCertificate {
+                    $vcfCertCa = Get-VCFCertificateAuthority
+                    if ($vcfCertCa.username -ne "$certAuthorityUser") {
+                        Write-Output "Configuring the Microsoft Certificate Authority in SDDC Manager using $($certAuthorityUser)..."
+                        Set-VCFMicrosoftCA -serverUrl $caServerUrl -username $certAuthorityUser -password $certAuthorityPass -templateName $certAuthorityTemplate | Out-Null
+                        Write-Output "Configuration of the Microsoft Certificate Authority in SDDC Manager using ($($certAuthorityUser)): SUCCESSFUL."
+                    } else {
+                        Write-Warning "Configuration of the Microsoft Certificate Authority in SDDC Manager using ($($certAuthorityUser)), already exists: SKIPPED."
+                    }
+                    Write-Output "Configuration a Microsoft Certificate Authority in SDDC Manager completed."
+                } Catch {
+                    $ErrorMessage = $_.Exception.Message
+                    Write-Output "Error was: $ErrorMessage."
+                }
+            } else {
+                Write-Error "Unable to authenticate to SDDC Manager ($($server)): PRE_VALIDATION_FAILED."
+            }
+        } else {
+            Write-Error "Unable to connect to Microsoft Certificate Authority ($certAuthorityFqdn)."
+        }
+    } else {
+        Write-Error "Unable to connect to SDDC Manager ($($server)): PRE_VALIDATION_FAILED."
+    }
+}
+
+Function gatherSddcInventory {
+    Param (
+        [Parameter (Mandatory = $true)] $domainType,
+        [Parameter (Mandatory = $true)] $workloadDomain
+    )
+
+    # Gathers deployment details from SDDC Manager.
+    $sddcMgr = Get-VCFManager
+    $sddcMgrVersion = $sddcMgr.version.split(".")[0]
+
+    $resourcesObject = @()
+    
+    # SDDC Manager
+    if ($domainType -eq "Management") {
+        $resourcesObject += [pscustomobject]@{
+            'fqdn'       = $sddcMgr.fqdn
+            'name'       = $sddcMgr.fqdn.split(".")[0]
+            'resourceId' = $sddcMgr.id
+            'type'       = "SDDC_MANAGER"
+        } 
+    }
+
+    # vCenter Server
+    if (([float]$sddcMgrVersion -ge [float]4) -AND ($domainType -eq "Management")) {
+        $domain = Get-VCFWorkloadDomain | Where-Object { $_.type -eq "MANAGEMENT" }
+        $vCenterServer = Get-VCFvCenter | Where-Object { $_.domain.id -eq $domain.id }
+    } else {
+        $domain = Get-VCFWorkloadDomain | Where-Object { $_.name -eq $workloadDomain }
+        $vCenterServer = Get-VCFvCenter | Where-Object { $_.domain.id -eq $domain.id }
+    }
+
+    Foreach ($vCenter in $vCenterServer) {
+        $resourcesObject += [pscustomobject]@{
+            'fqdn'       = $vCenter.fqdn
+            'name'       = $vCenter.fqdn.split(".")[0]
+            'resourceId' = $vCenter.id
+            'type'       = "VCENTER"
+        }
+    }
+
+    # NSX
+    if (([float]$sddcMgrVersion -ge [float]4) -OR ($isLegacyWldnsxT)) {
```

Comment:
```suggestion
    if (([float]$sddcMgrVersion -ge 4) -OR ($isLegacyWldnsxT)) {
```

Reply:
addressed in (881f078c8d28b1a0e966b304039b6594973f2314)